### PR TITLE
Add --signing-key (pkcs8) arg to in-toto-run

### PR DESCRIPTION
**Pull request by lukpueh**
*EDIT 2023/12/05: rename `--pkcs8` to `--signing-key`, and test with `--use-dsse`*
Fixes part of #533 (missing: --signing-key support in in-toto-record)

----

Unlike the existing `--key`, the here added `--signing-key` supports a standard file format (PKCS8/pem), which is consistent for all supported key types and eliminates the need for an explicitly passed `--key-type`.

This PR adds a warning that `--key` is deprecated and `--signing-key` should be used instead.

In preparation for the removal of  key file loading (plus decryption via passed or prompted password) in securesystemslib,
the functionality is added to in-toto in only a few lines of code. This is possible thanks to the modern securesystemslib CryptoSigner interface.

### CODE REVIEW

**Comment by lukpueh** [comment]
Note about CI failures:
- ~with-sslib-main fail is expected, because we use an unmerged sslib feature~
- ~lint fail is unrelated and will be fixed separately~

File: in_toto/common_args.py [hunk 1]
```diff
@@ -163,6 +163,18 @@ def parse_password_and_prompt_args(args):
     ),
 }
 
+PKCS8_ARGS = ["--pkcs8"]
```
Comment (lukpueh) [inline in_toto/common_args.py]: `pkcs8` is a bit technical. We could also call the arg:
- private-key
- signing-key
- key-file
- key-path
- ...


Any thoughts?

Comment (adityasaky) [reply in_toto/common_args.py]: I think I prefer `--signing-key`.

File: tests/test_in_toto_run.py [hunk 2]
```diff
@@ -364,6 +364,41 @@ def test_main_encrypted_key_but_no_pw(self):
         self.assert_cli_sys_exit(args, 1)
         self.assertFalse(os.path.exists(self.test_link_ed25519_enc))
 
+    def test_pkcs8(self):
+        """Test in-toto-run, sign link with pkcs8 key file for each algo."""
+        pems_dir = Path(__file__).parent / "pems"
+        args = ["-n", "foo", "-x", "--pkcs8"]
```
Comment (adityasaky) [inline tests/test_in_toto_run.py]: should this also test with `--use-dsse` for completeness?

**Review by adityasaky — APPROVED** [review]
LGTM, minus minor comments.

File: in_toto/common_args.py (hunk 1 above)
Comment (lukpueh) [reply in_toto/common_args.py]: Fair. Let's go with signing-key then.

**Pull request merged**